main.py

```diff
@@ -0,0 +1,241 @@
+                    else:
+                        break
+                self.data.append(single_file_records)
+        except : 
+            pass
+            
+        
+class CalculateTemperatureValues:
+    def __init__(self, file_obj):
+        self.data = file_obj
+        
+    def calculateTemperature(self, type_of_calculation):
+        if type_of_calculation == 'H':
+            return self._calculateTemperatureWithField('Max TemperatureC', True)
+        elif type_of_calculation == 'L': 
+            return self._calculateTemperatureWithField('Min TemperatureC', False)
+        elif type_of_calculation == 'Hu': 
+            return self._calculateTemperatureWithField('Max Humidity', True)
+        elif type_of_calculation == 'avg_L':
+            return self._calculateTemperatureWithField('Mean TemperatureC', False)
+        elif type_of_calculation == 'avg_H':
+            return self._calculateTemperatureWithField('Mean TemperatureC', True)
+        elif type_of_calculation == 'avg_Hu':
+            return self._calculateTemperatureWithField('Max Humidity', True)
+        else:
+            raise Exception("Incorrect Input")
+    
+    def _calculateTemperatureWithField(self, field, maxNumber):
+        latest_data = ""
+        required_data = ""
+        for single_file_record in self.data.data:
+            latest_data = self._calculateInfoFromFile(single_file_record, field, maxNumber)
+            if(not required_data):
+                required_data = latest_data
+                
+            if(maxNumber and latest_data["temperature"] > required_data["temperature"]):
+                required_data = latest_data 
+            else:
+                if latest_data["temperature"] < required_data["temperature"]:
+                    required_data = latest_data 
+                    
+            print(required_data)
+               
+        return required_data
+        
+    def _calculateInfoFromFile(self, single_file_record, field, maxNumber):
+        required_number = None
+        required_date = ''
+        for single_date_key in single_file_record.keys():
+                temperature = self._getFieldFromSingleDate(single_file_record, single_date_key, field)
+                if(not required_number): required_number = temperature
+                if temperature != '':
+                    if maxNumber and int(temperature) >= int(required_number):
+                        required_number = temperature
+                        required_date = single_date_key
+                    elif not maxNumber and  int(temperature) <= int(required_number):
+                        required_date = single_date_key
+                        required_number = temperature
+        return {
+            "temperature": required_number,
+            "date": required_date
+        }
+        
+    def _getFieldFromSingleDate(self, single_file_record, single_date_key, field):
+            return single_file_record[single_date_key][field]
```

Comment:
Can achieve in same line it is called, until it is not a straight forward logic and being used at multiple areas.